### PR TITLE
Us 5.4 room mapping

### PR DESCRIPTION
**Pull request by vtaing10**
## Description:
This PR implements the functionality to display directions between rooms located on the same floor.
Users can search for a destination room on their current floor, and the system calculates and displays the shortest path between the two rooms on the indoor map.

The routing logic for same-floor navigation was implemented and integrated with the indoor map so that the path is visually displayed to the user. Note that the shortest path is mathematically calculated. Even when sometimes a path seems visually shorter, it might not be mathematically.

## Related User Story:
[(Link the related task)
](https://github.com/Mini-Capstone-Fried-Chicken/Fried-Chicken/issues/23)

## Screenshot/Demo:
image

## How to test:
1. Run the application.
2. Navigate to the indoor map view of any building
3. Select a starting room on a floor.
4. Use the search bar to select another room on the same floor.
5. Confirm that the route is displayed visually on the indoor map.

## Type of Change:
- [x] Feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation update
- [ ] Tests
- [ ] Others

## Pull Request Checklist
- [ ] Flutter CI test passed
- [ ] Sonar test passed
- [ ] Codecov test passed
- [ ] Application runs without errors
- [ ] Feature tested locally on a high level
- [ ] Architecture conventions respected
- [ ] No API keys or secrets committed
- [ ] Document when necessary

## PR Review Checklist (dev)
- [ ] Code is readable and well structured
- [ ] No obvious bugs or edge cases missed
- [ ] Tests are adequate
- [ ] Naming and organization are clear
- [ ] Feature works as described
- [ ] No unnecessary complexity
- [ ] Follows project architecture

## PR Review Checklist (QA)
- [ ] Made sure to test the application fully
- [ ] No bugs created by this PR

### CODE REVIEW

File: lib/services/indoors_routing/indoor_dijkstra.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+      if (currentNodeId == null) break;
+      if (currentNodeId == endNodeId) break;
+
+      visited.add(currentNodeId);
+
+      final neighbors = adjacency[currentNodeId] ?? const [];
+      for (final edge in neighbors) {
+        final currentDistance = distances[currentNodeId]!;
+        final candidateDistance = currentDistance + edge.weightMeters;
```
Comment (maddie-luu) [inline lib/services/indoors_routing/indoor_dijkstra.dart]: How does this code works right now is that each loop scans through every node to find the next smallest distance. This is ok for our small indoor graphs, however it does pose scalability problems in the future. The time complexity would not be appropriate if we were to use larger building graphs

File: lib/services/indoors_routing/indoor_floor_graph_builder.dart
Comment (maddie-luu) [inline lib/services/indoors_routing/indoor_floor_graph_builder.dart]: Overall, this pr is quite complex, and can utilize more refactoring, code coverage in order to improve maintenance and quality

File: lib/services/indoors_routing/indoor_dijkstra.dart (hunk 1 above)
Comment (vtaing10) [reply lib/services/indoors_routing/indoor_dijkstra.dart]: I didn't think of that, nice catch! (And it is fixed)

File: lib/services/indoors_routing/indoor_same_floor_router.dart
[the remark this answers is not in the document]
Comment (vtaing10) [reply lib/services/indoors_routing/indoor_same_floor_router.dart]: Maybe a bit overkill? I split the the responsibilities intro concrete helpers instead. Thoughts?

[the remark this answers is not in the document]
Comment (maddie-luu) [reply lib/services/indoors_routing/indoor_same_floor_router.dart]: Concrete helpers are also a great way to go about this, good thinking!
